go/vt/vtgate/evalengine/compiler_asm.go

```diff
@@ -2725,6 +2726,72 @@ func (asm *assembler) Fn_TRIM2(col collations.TypedCollation) {
 	}, "FN TRIM VARCHAR(SP-2) VARCHAR(SP-1)")
 }
 
+func (asm *assembler) Fn_SUBSTRING2(tt sqltypes.Type, cs types.Charset, col collations.TypedCollation) {
+	asm.adjustStack(-1)
+	asm.emit(func(env *ExpressionEnv) int {
+		str := env.vm.stack[env.vm.sp-2].(*evalBytes)
+		pos := env.vm.stack[env.vm.sp-1].(*evalInt64)
+
+		end := int64(charset.Length(cs, str.bytes))
+		if pos.i < 0 {
+			pos.i += end + 1
+		}
+		if pos.i < 1 || pos.i > end {
+			str.tt = int16(tt)
```

Comment:
nit: `str.tt = int16(tt)` is present in both the `if` condition and outside it. This can be moved above so that we don't need to repeat it.

Reply:
Fixed!